pkg/kubelet/cm/dra/manager.go

```diff
@@ -208,43 +197,54 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 		}
 
 		// Skip calling NodeUnprepareResource if other pods are still referencing it
-		if len(claimInfo.podUIDs) > 1 {
+		if len(claimInfo.PodUIDs) > 1 {
+			// It OK not checkpoint the deletePodReference() here and below because if it
+			// missing after kubelet restart it means that we have never returned success from the
+			// UnprepareResources() call for this pod and it never deleted by the kubelet.
+			// Eventaully we capture all deletePodReference() calls at the end of the UnprepareResources().
 			claimInfo.deletePodReference(pod.UID)
 			continue
 		}
 
 		// Call NodeUnprepareResource only for the last pod that references the claim
-		client, err := dra.NewDRAPluginClient(claimInfo.driverName)
+		client, err := dra.NewDRAPluginClient(claimInfo.DriverName)
 		if err != nil {
-			return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", claimInfo.driverName, err)
+			return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", claimInfo.DriverName, err)
 		}
 
 		response, err := client.NodeUnprepareResource(
 			context.Background(),
-			claimInfo.namespace,
-			claimInfo.claimUID,
-			claimInfo.claimName,
-			claimInfo.cdiDevices)
+			claimInfo.Namespace,
+			claimInfo.ClaimUID,
+			claimInfo.ClaimName,
+			claimInfo.CdiDevices)
 		if err != nil {
 			return fmt.Errorf(
 				"NodeUnprepareResource failed, pod: %s, claim UID: %s, claim name: %s, CDI devices: %s, err: %+v",
-				pod.Name,
-				claimInfo.claimUID,
-				claimInfo.claimName,
-				claimInfo.cdiDevices, err)
+				pod.Name, claimInfo.ClaimUID, claimInfo.ClaimName, claimInfo.CdiDevices, err)
 		}
 
+		klog.V(3).InfoS("NodeUnprepareResource succeeded", "response", response)
+
 		// Delete last pod UID only if NodeUnprepareResource call succeeds.
 		// This ensures that status manager doesn't enter termination status
 		// for the pod. This logic is implemented in the m.PodMightNeedToUnprepareResources
 		// and in the claimInfo.hasPodReference.
 		claimInfo.deletePodReference(pod.UID)
 
-		klog.V(3).InfoS("NodeUnprepareResource succeeded", "response", response)
-		// delete resource from the cache
-		m.cache.delete(claimInfo.claimName, pod.Namespace)
+		// Delete resource from the cache
+		m.cache.delete(claimInfo.ClaimName, pod.Namespace)
+		// We checkpoint to keep it consistent with cache.add()
```

Comment:
// Checkpoint to reduce redundant calls to NodeUnPrepareResource() after a kubelet restart.